osu.Game/Graphics/UserInterface/TwoLayerButton.cs

```diff
@@ -239,5 +199,69 @@ protected override bool OnClick(InputState state)
 
             return base.OnClick(state);
         }
+
+        private class IconBeatSyncedContainer : BeatSyncedContainer

```

Comment:
just call it `BouncingIcon` or something. You don't need to/shouldn't give it a general name when it's private single-use.